diff --git a/openstack_dashboard/urls.py b/openstack_dashboard/urls.py
index 64ef605..61ad1dc 100644
--- a/openstack_dashboard/urls.py
+++ b/openstack_dashboard/urls.py
@@ -27,13 +27,38 @@ from django.conf.urls.static import static
 from django.conf import settings
 from django.contrib.staticfiles.urls import staticfiles_urlpatterns
 
-import horizon
 
+# do some monkey patching
+from tukey.shibboleth_auth import patch_openstack_middleware_get_user
+import openstack_auth.views
+from django.shortcuts import redirect
+openstack_auth.views.login = lambda req: redirect('/federated/login/')
+
+patch_openstack_middleware_get_user()
+
+import horizon
 
 urlpatterns = patterns('',
+
+    url(r'^$', 'tukey.content.views.page', name='home'),
+    url(r'^console/', 'django_openid_auth.views.login_begin'),
+    url(r'^federated/', include('django_openid_auth.urls', namespace='openid')),
+#start
+#    url(r'^files/', include('tukey.files.urls', namespace='files')),
+#    url(r'^tukey_admin/', include('tukey.tukey_admin.urls', namespace='tukey_admin')),
+    url(r'^status/', include('tukey.status.urls', namespace='status')),
+    url(r'^publicdata/', include('tukey.datasets.urls', namespace='datasets')),
+    url(r'^keyservice/', include('tukey.keyservice.urls', namespace='keyservice')),
+    url(r'^osdcquery/', include('tukey.osdcquery.urls', namespace='osdquery')),
+    url(r'', include('tukey.webforms.urls')),
+    url(r'', include(horizon.urls)),
+
     url(r'^$', 'openstack_dashboard.views.splash', name='splash'),
     url(r'^auth/', include('openstack_auth.urls')),
-    url(r'', include(horizon.urls))
+
+    #url(r'^switch/(?P<tenant_id>[^/]+)/$', 'switch', name='switch_tenants')
+
+
 )
 
 # Development static app and project media serving using the staticfiles app.
@@ -44,7 +69,16 @@ urlpatterns += staticfiles_urlpatterns()
 # path. Production media should NOT be served by Django.
 urlpatterns += static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)
 
+
+# content has to be last
+
 if settings.DEBUG:
     urlpatterns += patterns('',
-        url(r'^500/$', 'django.views.defaults.server_error')
+        url(r'^500/$', 'django.views.defaults.server_error'),
+        url(r'', include('tukey.content.urls', namespace='content'))
     )
+else:
+    urlpatterns += patterns('',
+        url(r'', include('tukey.content.urls', namespace='content'))
+    )
+
